proto/proto/store/rpc.proto

```diff
@@ -103,71 +103,123 @@ message StoreStatus {
 // GET ACCOUNT PROOF
 // ================================================================================================
 
-// Returns the latest state proof of the specified accounts.
+// Returns the latest state proof of the specified account.
 message AccountProofRequest {
     // Request the details for a public account.
-    message AccountDetailsRequest {
+    message AccountDetailRequest {
         // Represents a storage slot index and the associated map keys.
-        message StorageRequest {
-            // Storage slot index ([0..255])
-            uint32 storage_slot_index = 1;
-
-            // A list of map keys (Digests) associated with this storage slot.
-            repeated primitives.Digest map_keys = 2;
+        message StorageMapDetailRequest {
+            // Indirection required for use in `oneof {..}` block.
+            message MapKeys {
+                // A list of map keys associated with this storage slot.
+                repeated primitives.Digest map_keys = 1;
+            }
+            // Storage slot index (`[0..255]`).
+            uint32 slot_index = 1;
+
+            oneof slot_data {
+                // Request to return all storage map data. If the number exceeds a threshold of 1000 entries,
+                // the response will not contain them but must be requested separately.
+                bool all_entries = 2;
+
+                // A list of map keys associated with the given storage slot identified by `slot_index`.
+                MapKeys map_keys = 3;
+            }
         }
 
-        // Account code commitment corresponding to the last-known `AccountCode` for the requested
-        // account. The response will include only code that is known.
+        // Last known code commitment to the client. The response will include account code
+        // only if its commitment is different from this value or if the value is not present.
         optional primitives.Digest code_commitment = 1;
 
-        // List of storage requests for this account.
-        repeated StorageRequest storage_requests = 2;
+        // Last known asset vault commitment to the client. The response will include asset vault data
+        // only if its commitment is different from this value or if the value is not present.
+        // If the number of to-be-returned asset entries exceed a threshold, they have to be requested
```

Comment:
I think I'd make this consistent with the account code request and not return anything when the value is `None`